### ISSUE TITLE
Improve Duplicate File UX

### ISSUE THREAD

**Issue by mlucool** [jupyterlab/jupyterlab#8132]
Duplicating notebooks can be improved to be more inline with other editors.

Today in the file browser, context menus->duplicate  `notebook.ipynb` creates `notebook-Copy1.ipynb` and keeps the original notebook selected.

Proposal:
1. Clicking duplicates opens up the rename popup with the text `notebook-Copy1.ipynb` and it all selected. This lets the user push enter to create or type to rename.
2. Duplicating a notebook opens it after creation.
Nice to have:
3. Add a duplicate option to the tab's context menu.

1/2 mirrors copy/paste of files in Jetbrains IDEs (but retains the suggested name convention). VSCode does not have a popup, but does open the file.

**Comment by krassowski** [comment]
I agree that UX for duplication needs improvements! Very often I duplicate a notebook and rename the wrong one because of the selection staying on the original one (and because oftentimes I do not see the full names in the file browser). Then I carry on working in a wrong copy and find myself frustrated that I had overwritten my work because the selection stayed on the original notebook and I did not notice. Proposal (1) sounds like a very good idea!

However, I think that opening the notebook after duplication (proposal 2) is not the best solution - I often have memory limitations and duplicate large notebooks in order two split those in the two; in this use case be definition opening both notebooks causes a significant strain on the browser (especially given #4292).

I also support the proposal (3) - I tried to duplicate a notebook from the tab's context menu a few days ago, I just expected this option to be there.

**Comment by mlucool** [comment]
> especially given #4292

Assuming this is solved, would you prefer  2? As an FYI @saulshanabrook is working to solve this. If you have more data for him to benchmark (https://github.com/jupyterlab/jupyterlab/pull/8020) - I'm sure he'd appreciate it!

**Comment by krassowski** [comment]
Yes, I know, I did [some](https://github.com/jupyterlab/jupyterlab/issues/4292#issuecomment-521853698) [benchmarks](https://github.com/jupyterlab/jupyterlab/issues/4292#issuecomment-521856529) a while ago too. Once this is fixed I would not have a strong opinion on point two (nor preference).

**Comment by jtpio** [comment]
Also it would be useful to be able to hold <kbd>Ctrl</kbd> and drag and drop a file to duplicate it (similar to file browsers on a desktop).

**Comment by tgeorgeux** [comment]
> Today in the file browser, context menus->duplicate notebook.ipynb creates notebook-Copy1.ipynb and keeps the original notebook selected.

@mlucool and @krassowski which version of JupyterLab are you running? I just tried to duplicate this behavior in JupyterLab 2.0.1 and JupyterLab 2.1.0. In both versions I had the following result:
1. Right-click untitled.ipynb and click duplicate from the context menu.
2. untitled-copy1.ipynb was created and selected. 

Here's a gif of my interaction.
![8132video](https://user-images.githubusercontent.com/22109376/79514371-e80dee00-7ffa-11ea-9a97-e0a69dd1a2f1.gif)

> 1. Clicking duplicates opens up the rename popup with the text notebook-Copy1.ipynb and it all selected. This lets the user push enter to create or type to rename.

I can see arguments in favor of opening a naming dialogue and against it, does anybody else have a strong opinion on this? 

Would just selecting the notebook copy on duplication be a step in the right direction?

> However, I think that opening the notebook after duplication (proposal 2) is not the best solution - I often have memory limitations and duplicate large notebooks in order two split those in the two; in this use case be definition opening both notebooks causes a significant strain on the browser (especially given #4292).

I agree we need to fix the lazy-loading of notebooks before we launch a notebook on duplication.

> 3. Add a duplicate option to the tab's context menu.

In this case, I would contradict my previous statement, if a file that's already open is being duplicated it seems less risky to assume somebody wants to launch a copy of it.

**Comment by mlucool** [comment]
I was running `1.2.x`. I thought I tried something more recent, but it seems the binder link in the README is broken.

I think the dialog would encourage most people to choose a name. I personally am in favor.

> I agree we need to fix the lazy-loading of notebooks before we launch a notebook on duplication.

@saulshanabrook is working on that :)
